### ISSUE TITLE
Dozens of Amazon S3 Errors in Log: PHP Invalid argument foreach() at AmazonS3.php#295	

### ISSUE THREAD

**Issue by chriswayg** [nextcloud/server#9403]
### Steps to reproduce
1. External Storage app enabled
2. Create an Amazon S3 bucket in Nextcloud
3. Add a few files on the desktop and sync
4. Check the log

### Actual behaviour
Dozens of Amazon S3 Errors in Log:

Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:34:35+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:34:34+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:34:33+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:34:33+0800
...

## Server configuration detail

**Operating system:** Linux 4.9.0-6-amd64 #1 SMP Debian 4.9.88-1 (2018-04-29) x86_64

**Webserver:** nginx/1.13.12 (fpm-fcgi)

**Database:** mysql 10.2.14

**PHP version:** 7.2.4
Modules loaded: Core, date, libxml, openssl, pcre, sqlite3, zlib, ctype, curl, dom, fileinfo, filter, ftp, hash, iconv, json, mbstring, SPL, PDO, pdo_sqlite, session, posix, readline, Reflection, standard, SimpleXML, Phar, tokenizer, xml, xmlreader, xmlwriter, mysqlnd, cgi-fcgi, apcu, gd, gettext, imagick, imap, intl, ldap, mailparse, memcached, pcntl, pdo_mysql, redis, soap, sockets, sodium, xmlrpc, zip, Zend OPcache

**Nextcloud version:** 13.0.2 - 13.0.2.1

**Updated from an older Nextcloud/ownCloud or fresh install:** 

**Where did you install Nextcloud from:** 
https://github.com/mailcow/mailcow-dockerized/blob/master/helper-scripts/nextcloud.sh

<details><summary>Signing status</summary>

Array
</details>

<details><summary>List of activated apps</summary>

```
Enabled:
 - activity: 2.6.1
 - comments: 1.3.0
 - dav: 1.4.6
 - federatedfilesharing: 1.3.1
 - federation: 1.3.0
 - files: 1.8.0
 - files_external: 1.4.1
 - files_pdfviewer: 1.2.1
 - files_sharing: 1.5.0
 - files_texteditor: 2.5.1
 - files_trashbin: 1.3.0
 - files_versions: 1.6.0
 - files_videoplayer: 1.2.0
 - firstrunwizard: 2.2.1
 - gallery: 18.0.0
 - issuetemplate: 0.3.0
 - logreader: 2.0.0
 - lookup_server_connector: 1.1.0
 - nextcloud_announcements: 1.2.0
 - notifications: 2.1.2
 - oauth2: 1.1.0
 - password_policy: 1.3.0
 - provisioning_api: 1.3.0
 - serverinfo: 1.3.0
 - sharebymail: 1.3.0
 - survey_client: 1.1.0
 - systemtags: 1.3.0
 - theming: 1.4.1
 - twofactor_backupcodes: 1.2.3
 - updatenotification: 1.3.0
 - user_external: 0.4
 - workflowengine: 1.3.0
Disabled:
 - admin_audit
 - encryption
 - user_ldap

```
</details>

<details><summary>Configuration (config/config.php)</summary>

```
{
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "localhost",
        "cloud.lightinasia.site"
    ],
    "datadirectory": "***REMOVED SENSITIVE VALUE***",
    "overwrite.cli.url": "http:\/\/localhost",
    "dbtype": "mysql",
    "version": "13.0.2.1",
    "dbname": "***REMOVED SENSITIVE VALUE***",
    "dbhost": "***REMOVED SENSITIVE VALUE***",
    "dbport": "",
    "dbtableprefix": "nc_",
    "mysql.utf8mb4": true,
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "installed": true,
    "instanceid": "***REMOVED SENSITIVE VALUE***",
    "redis": {
        "host": "***REMOVED SENSITIVE VALUE***",
        "port": 6379,
        "timeout": 0
    },
    "memcache.locking": "\\OC\\Memcache\\Redis",
    "memcache.local": "\\OC\\Memcache\\Redis",
    "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
    "overwritewebroot": "\/",
    "overwritehost": "cloud.lightinasia.site",
    "overwriteprotocol": "https",
    "mail_smtpmode": "smtp",
    "mail_smtpauthtype": "LOGIN",
    "mail_from_address": "***REMOVED SENSITIVE VALUE***",
    "mail_domain": "***REMOVED SENSITIVE VALUE***",
    "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpport": "588",
    "user_backends": [
        {
            "arguments": [
                "{dovecot:143\/imap\/tls\/novalidate-cert}"
            ],
            "class": "OC_User_IMAP"
        }
    ]
}
```
</details>

**Are you using external storage, if yes which one:** 
Amazon S3 Asia/Singapore

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

## Client configuration

**Browser:** Mozilla/5.0 (Macintosh; Intel Mac OS X 10.12; rv:59.0) Gecko/20100101 Firefox/59.0

**Operating system:** 

## Logs

<details><summary>Browser log</summary>

```
Insert your webserver log here 
```
</details>

<details><summary>Nextcloud log</summary>

```
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:32:07+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:32:07+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:32:07+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:32:07+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:32:03+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:32:03+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:32:03+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:32:03+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:31:59+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:31:59+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:31:59+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:31:59+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:30:53+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:30:53+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:30:41+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:30:41+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:30:41+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:30:06+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:30:05+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:30:05+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:30:05+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:29:07+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:29:06+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:29:06+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:29:06+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:28:06+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:28:05+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:28:05+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:28:05+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:27:54+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:27:06+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:27:06+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:27:05+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:27:05+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:26:06+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:26:05+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:26:05+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:26:05+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:25:08+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:25:07+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:25:07+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:25:07+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:24:06+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:24:06+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:24:05+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:24:05+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:23:06+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:23:06+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:23:05+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:23:05+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:22:06+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:22:06+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:22:05+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:22:05+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:21:06+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:21:06+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:21:05+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:21:05+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:20:05+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:20:04+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:20:03+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:20:03+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:16:36+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:16:36+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:16:35+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:16:35+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:15:37+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:15:36+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:15:36+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:15:36+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:14:37+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:14:36+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:14:36+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:14:36+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:13:37+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:13:36+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:13:36+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:13:36+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:12:36+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:12:36+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:12:35+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:12:35+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:12:33+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:12:33+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:12:33+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:12:33+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:12:24+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:12:24+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:12:24+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:12:24+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:12:22+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:12:22+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:12:22+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:12:22+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:12:03+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:12:03+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:12:03+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:12:03+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:12:00+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:12:00+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:12:00+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:11:59+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:11:48+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:11:47+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:11:47+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:11:46+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:11:46+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:11:45+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:11:44+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:11:44+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:11:44+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:11:40+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:11:24+0800
Error	PHP	Invalid argument supplied for foreach() at /web/nextcloud/apps/files_external/lib/Lib/Storage/AmazonS3.php#295	2018-05-05T20:10:35+0800
```
</details>

<details><summary>Local sync protocol</summary>

```
  admin@cloud.lightinasia.site,       Sat May 5 12:38:46 2018 GMT,/ExternalS3/Test/accessories/download (5).jpeg,You created ExternalS3/Nextcloud.mp4, ExternalS3/Test/accessories/download.jpeg, ExternalS3/Test/accessories/download (2).jpeg, ExternalS3/Test/accessories/download (6).jpeg and ExternalS3/Test/accessories/download (5).jpeg,                                                       
  admin@cloud.lightinasia.site,       Sat May 5 12:38:46 2018 GMT,  /ExternalS3/Test/accessories,You created ExternalS3/Test/accessories/download (1).jpeg, ExternalS3/Test/accessories/download (4).jpeg, ExternalS3/Test/accessories/download (3).jpeg and ExternalS3/Test/accessories,                                                       
  admin@cloud.lightinasia.site,       Sat May 5 12:38:04 2018 GMT,              /ExternalS3/Test,                                           You renamed ExternalS3/untitled folder to ExternalS3/Test,                                                       
  admin@cloud.lightinasia.site,       Sat May 5 12:11:47 2018 GMT,/ExternalS3/Photos/Hummingbird.jpg,You created ExternalS3/untitled folder, ExternalS3/Nextcloud Manual.pdf, ExternalS3/Photos/Nut.jpg, ExternalS3/Photos/Coast.jpg and ExternalS3/Photos/Hummingbird.jpg,                                                       
  admin@cloud.lightinasia.site,       Sat May 5 12:11:41 2018 GMT,         /ExternalS3/Documents,You created ExternalS3/Photos, ExternalS3/Documents/About.odt, ExternalS3/Documents/About.txt and ExternalS3/Documents,                                                       
  admin@cloud.lightinasia.site,       Sat May 5 12:00:04 2018 GMT,                              ,                                                                       You created calendar Personal,                                                       
  admin@cloud.lightinasia.site,       Sat May 5 12:00:04 2018 GMT,                /Nextcloud.mp4,      You created Photos/Coast.jpg, Photos/Hummingbird.jpg, Photos/Nut.jpg, Photos and Nextcloud.mp4,                                                       
  admin@cloud.lightinasia.site,       Sat May 5 12:00:03 2018 GMT,         /Nextcloud Manual.pdf,            You created Documents/About.txt, Documents/About.odt, Documents and Nextcloud Manual.pdf,                                                       

...
05/05/2018 8:34:37 PM,                                 ExternalS3/Nextcloud Manual.pdf,                     Nextcloud,       Uploaded,    4.3 MB
05/05/2018 8:33:55 PM,                               ExternalS3/Photos/Hummingbird.jpg,                     Nextcloud,     Downloaded,    572 KB
05/05/2018 8:33:55 PM,                                       ExternalS3/Photos/Nut.jpg,                     Nextcloud,     Downloaded,    933 KB
05/05/2018 8:33:54 PM,                                     ExternalS3/Photos/Coast.jpg,                     Nextcloud,     Downloaded,    801 KB
05/05/2018 8:33:54 PM,                                  ExternalS3/Documents/About.txt,                     Nextcloud,     Downloaded,      1 KB
05/05/2018 8:33:54 PM,                                  ExternalS3/Documents/About.odt,                     Nextcloud,     Downloaded,     76 KB
05/05/2018 8:33:53 PM,                                               ExternalS3/Photos,                     Nextcloud,     Downloaded,          
05/05/2018 8:33:53 PM,                                            ExternalS3/Documents,                     Nextcloud,     Downloaded,          
05/05/2018 8:33:53 PM,                                                      ExternalS3,                     Nextcloud,     Downloaded,          
05/05/2018 8:30:48 PM,                                                      ExternalS3,                     Nextcloud,        Deleted,          
05/05/2018 8:11:48 PM,                                       ExternalS3/Photos/Nut.jpg,                     Nextcloud,       Uploaded,    933 KB
05/05/2018 8:11:47 PM,                                     ExternalS3/Photos/Coast.jpg,                     Nextcloud,       Uploaded,    801 KB
05/05/2018 8:11:47 PM,                               ExternalS3/Photos/Hummingbird.jpg,                     Nextcloud,       Uploaded,    572 KB
05/05/2018 8:11:45 PM,                                               ExternalS3/Photos,                     Nextcloud,       Uploaded,          
05/05/2018 8:11:41 PM,                                  ExternalS3/Documents/About.odt,                     Nextcloud,       Uploaded,     76 KB
05/05/2018 8:11:41 PM,                                  ExternalS3/Documents/About.txt,                     Nextcloud,       Uploaded,      1 KB
05/05/2018 8:11:41 PM,                                            ExternalS3/Documents,                     Nextcloud,       Uploaded,          
05/05/2018 8:11:24 PM,                                                      ExternalS3,                     Nextcloud,     Downloaded,              
```
</details>

**Comment by MorrisJobke** [comment]
cc @icewind1991 @rullzer

**Comment by MorrisJobke** [comment]
Fix seems to be in #8842
